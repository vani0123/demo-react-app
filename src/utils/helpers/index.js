import StringHelper from './stringHelper/stringHelper';
import ObjectHelper from './objectHelper/objectHelper';
import ArrayHelper from './arrayHelper/arrayHelper';

export { StringHelper, ObjectHelper, ArrayHelper };
