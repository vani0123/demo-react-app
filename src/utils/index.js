/**
 * Main export for utilities that will be used across application.
 * Utilities are static classes that are used for providing standalone utility methods.
 * All the third party utility wrappers must be place here.
 *
 * eg -
 * import { CommonUtil, Transform } from './utils';
 * import * as utils from './utils';
 */
import CommonUtil from './common/common';
import Transform from './transform/transform';
import LoggingUtil from './logging/logging';
import FormatUtil from './format/format';
import * as helpers from './helpers';

export {
  CommonUtil, Transform, LoggingUtil, FormatUtil, helpers
};
