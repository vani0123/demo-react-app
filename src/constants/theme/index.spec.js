import THEME_CONSTANTS from './index';
import { Transform } from '../../utils';

const variablesDefault = jest.mock('../../styles/themes/_default.module.scss');
const variablesDark = jest.mock('../../styles/themes/_dark.module.scss');

const transformedVariablesDefault = Transform.serialize(variablesDefault);
const transformedVariablesDark = Transform.serialize(variablesDark);

describe('THEME_CONSTANTS', () => {
  it('should create default theme from scss variable', () => {
    expect(THEME_CONSTANTS.DEFAULT(transformedVariablesDefault)).toBeDefined();
  });

  it('should create dark theme from scss variable', () => {
    expect(THEME_CONSTANTS.DARK(transformedVariablesDark)).toBeDefined();
  });

  it('should have valid snapshot', () => {
    expect(
      THEME_CONSTANTS.DEFAULT(transformedVariablesDefault)
    ).toMatchSnapshot();
    expect(THEME_CONSTANTS.DARK(transformedVariablesDark)).toMatchSnapshot();
  });
});
