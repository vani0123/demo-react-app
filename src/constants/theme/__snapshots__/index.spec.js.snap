// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THEME_CONSTANTS should have valid snapshot 1`] = `
Object {
  "breakpoints": Object {
    "values": Object {
      "lg": 1200,
      "md": 992,
      "sm": 768,
      "xl": 1400,
      "xs": 0,
    },
  },
  "palette": Object {
    "action": Object {
      "activatedOpacity": NaN,
      "active": undefined,
      "disabled": undefined,
      "disabledBackground": undefined,
      "disabledOpacity": NaN,
      "focus": undefined,
      "focusOpacity": NaN,
      "hover": undefined,
      "hoverOpacity": NaN,
      "selected": undefined,
      "selectedOpacity": NaN,
    },
    "background": Object {
      "default": undefined,
      "paper": undefined,
    },
    "error": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "grey": Object {
      "100": undefined,
      "200": undefined,
      "300": undefined,
      "400": undefined,
      "50": undefined,
      "500": undefined,
      "600": undefined,
      "700": undefined,
      "800": undefined,
      "900": undefined,
      "A100": undefined,
      "A200": undefined,
      "A400": undefined,
      "A700": undefined,
    },
    "info": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "primary": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "secondary": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "success": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "text": Object {
      "disabled": undefined,
      "divider": undefined,
      "hint": undefined,
      "icon": undefined,
      "primary": undefined,
      "secondary": undefined,
    },
    "warning": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
  },
  "shape": Object {
    "borderRadius": NaN,
  },
  "typography": Object {
    "body1": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "body2": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "button": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "textTransform": undefined,
    },
    "caption": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
    },
    "fontFamily": undefined,
    "fontSize": NaN,
    "fontWeightBold": undefined,
    "fontWeightLight": undefined,
    "fontWeightMedium": undefined,
    "fontWeightRegular": undefined,
    "h1": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h2": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h3": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h4": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h5": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h6": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "htmlFontSize": NaN,
    "overline": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "textTransform": undefined,
    },
    "subtitle1": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "subtitle2": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
  },
  "zIndex": Object {
    "appBar": 1250,
    "drawer": 1200,
    "mobileStepper": 1000,
    "modal": 1300,
    "snackbar": 1400,
    "speedDial": 1050,
    "tooltip": 1500,
  },
}
`;

exports[`THEME_CONSTANTS should have valid snapshot 2`] = `
Object {
  "breakpoints": Object {
    "values": Object {
      "lg": 1200,
      "md": 992,
      "sm": 768,
      "xl": 1400,
      "xs": 0,
    },
  },
  "palette": Object {
    "action": Object {
      "activatedOpacity": NaN,
      "active": undefined,
      "disabled": undefined,
      "disabledBackground": undefined,
      "disabledOpacity": NaN,
      "focus": undefined,
      "focusOpacity": NaN,
      "hover": undefined,
      "hoverOpacity": NaN,
      "selected": undefined,
      "selectedOpacity": NaN,
    },
    "background": Object {
      "default": undefined,
      "paper": undefined,
    },
    "error": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "grey": Object {
      "100": undefined,
      "200": undefined,
      "300": undefined,
      "400": undefined,
      "50": undefined,
      "500": undefined,
      "600": undefined,
      "700": undefined,
      "800": undefined,
      "900": undefined,
      "A100": undefined,
      "A200": undefined,
      "A400": undefined,
      "A700": undefined,
    },
    "info": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "primary": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "secondary": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "success": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
    "text": Object {
      "disabled": undefined,
      "divider": undefined,
      "hint": undefined,
      "icon": undefined,
      "primary": undefined,
      "secondary": undefined,
    },
    "warning": Object {
      "contrastText": undefined,
      "dark": undefined,
      "light": undefined,
      "main": undefined,
    },
  },
  "shape": Object {
    "borderRadius": NaN,
  },
  "typography": Object {
    "body1": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "body2": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "button": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "textTransform": undefined,
    },
    "caption": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
    },
    "fontFamily": undefined,
    "fontSize": NaN,
    "fontWeightBold": undefined,
    "fontWeightLight": undefined,
    "fontWeightMedium": undefined,
    "fontWeightRegular": undefined,
    "h1": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h2": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h3": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h4": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h5": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "h6": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "htmlFontSize": NaN,
    "overline": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "textTransform": undefined,
    },
    "subtitle1": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
    "subtitle2": Object {
      "fontFamily": undefined,
      "fontSize": undefined,
      "fontWeight": undefined,
      "letterSpacing": undefined,
      "lineHeight": undefined,
    },
  },
  "zIndex": Object {
    "appBar": 1250,
    "drawer": 1200,
    "mobileStepper": 1000,
    "modal": 1300,
    "snackbar": 1400,
    "speedDial": 1050,
    "tooltip": 1500,
  },
}
`;
